web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyServiceTest.java

```diff
@@ -0,0 +1,159 @@
+package edu.unc.lib.boxc.web.services.processing;
+
+import org.apache.commons.csv.CSVRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static edu.unc.lib.boxc.web.services.processing.SingleUseKeyService.CSV_HEADERS;
+import static edu.unc.lib.boxc.web.services.processing.SingleUseKeyService.DAY_MILLISECONDS;
+import static edu.unc.lib.boxc.web.services.utils.CsvUtil.createCsvPrinter;
+import static edu.unc.lib.boxc.web.services.utils.CsvUtil.parseCsv;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @author snluong
+ */
+public class SingleUseKeyServiceTest {
+    @TempDir
+    public Path tmpFolder;
+    private Path csvPath;
+
+    private static final String UUID_1 = "f277bb38-272c-471c-a28a-9887a1328a1f";
+    private static final String UUID_2 = "ba70a1ee-fa7c-437f-a979-cc8b16599652";
+    private static final String UUID_3 = "83c2d7f8-2e6b-4f0b-ab7e-7397969c0682";
+    private static final String UUID_TEST = "0e33ad0b-7a16-4bfa-b833-6126c262d889";
+    private final String[] ids = {UUID_1, UUID_2, UUID_3};
+    private SingleUseKeyService singleUseKeyService;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        csvPath = tmpFolder.resolve("singleUseKey");
+        singleUseKeyService = new SingleUseKeyService();
+        singleUseKeyService.setCsvPath(csvPath);
+    }
+
+    @Test
+    public void testGenerateNoCsvExists() throws IOException {
+        var key = singleUseKeyService.generate(UUID_TEST);
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertCsvContainsCorrectEntry(newRecords, UUID_TEST, key, currentMilliseconds);
+    }
+
+    @Test
+    public void testGenerateMultipleCallsForDifferentIds() throws IOException {
```

Comment:
I ran this test and inspected the CSV file it generates, and it looks like CsvUtil needs an adjustment to prevent it from adding headers for already existing files. At the moment the CSV from this test is:
```
UUID,Access Key,Expiration Timestamp
f277bb38-272c-471c-a28a-9887a1328a1f,fa2aba6ed5aa4718af1d497778cfaf792593d477356f3,1704575328754
UUID,Access Key,Expiration Timestamp
ba70a1ee-fa7c-437f-a979-cc8b16599652,b60c1430637f46d8b765dc2885d219d52593d4a01cbc7,1704575328775
UUID,Access Key,Expiration Timestamp
83c2d7f8-2e6b-4f0b-ab7e-7397969c0682,44fc9960d7434cbbb307f504f10f00f92593d4a0c11df,1704575328776
```

To make sure we don't end up in a situation like this later, we should also add an assertion on the number of records in the CSV.